### ISSUE TITLE
ColorProps types not working in hoc

### ISSUE THREAD

**Issue by nklaasse** [styled-system/styled-system#1206]
**Describe the bug**
Using `ColorProps` types within a hoc seems not to work as expected. All other api props types such as `DisplayProps`, `VerticalAlignProps`, `SpaceProps` etc. are working as expected.

**To Reproduce**
https://codesandbox.io/s/elegant-gates-hsot7

```
Type '{ size: string | string[]; color?: string | number | symbol | (string | number | symbol | null)[] | { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; } | null | undefined; ... 4 more ...; verticalAlign?: string | ... 4 more ... | undefined; }' is not assignable to type 'Pick<Pick<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "hidden" | "style" | "title" | "key" | "defaultChecked" | "defaultValue" | "suppressContentEditableWarning" | ... 247 more ... | "onTransitionEndCapture"> & { ...; } & TextProps, "hidden" | ... 259 more ... | "verticalAlign"> & Partial...'.
  Types of property 'color' are incompatible.
    Type 'string | number | symbol | (string | number | symbol | null)[] | { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; } | null | undefined' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.
      Type 'null' is not assignable to type 'string | (string & (string | number | symbol | null)[]) | (string & { [x: string]: string | number | symbol | undefined; [x: number]: string | number | symbol | undefined; }) | undefined'.ts(2322)
```

**Comment by g-a-v-i-n** [comment]
Encountering this as well

**Comment by smolak** [comment]
I believe I encountered the same issue. I tried non-TS code to see if I can set responsive color values (and if it will work). It does, but the compiler still highlights the code as wrongly used. See: https://codesandbox.io/s/styled-system-playground-n8n9v

tl;dr - I can't go with TextColorProps interface. I had to create my own and not allow to have color to be set in a responsive way.

**Comment by ashr81** [comment]
I am facing this issue because of typescript `types` conflict between `react` and `styled-system` types.
`@types/react` also has `color` property set to `string | undefined` which does not match with `styled-system` types.
Did you find any resolution for this issue @nklaasse , @smolak , @g-a-v-i-n ?

**Comment by matthova** [comment]
@smolak could you share your solution? Good call-out. I'm guessing in most cases, a responsive color isn't necessary, so this workaround seems like an okay compromise.

**Comment by smolak** [comment]
@matthova I stumbled upon this [stackoverflow question](https://stackoverflow.com/questions/53711454/styled-system-props-typing-with-typescript) and used solution from there:

```ts
import { color as ssColor, ColorProps as SSColorProps, TextColorProps, compose, system } from 'styled-system';

// Styled-system patch for the color prop fixing "Types of property 'color' are incompatible"
// when applying props to component that extend ColorProps.
export interface ColorProps extends Omit<SSColorProps, 'color'> {
    textColor?: TextColorProps['color'];
}

export const color = compose(
    ssColor,
    system({
        // Alias color as textColor
        textColor: {
            property: 'color',
            scale: 'colors',
        },
    })
);
```

What it does is exposes the `textColor` property on anything that should have it. The `color` is still available, but IDE will not suggest it (and will TS error, but will compile, since it is still used by the styled system).

**Comment by matthova** [comment]
Thanks @smolak will try this out

**Comment by smolak** [comment]
@matthova sure thing. If you'll have any questions, let me know.

**Comment by remco-zwartepoorte** [comment]
Adding the `as` prop fixed it for me. But don't ask me why 🤡 
```import React from 'react';
import styled from 'styled-components';
import {
    border,
    BorderProps,
    color,
    ColorProps,
    compose,
    flexbox,
    FlexboxProps,
    layout,
    LayoutProps,
    position,
    PositionProps,
    shadow,
    ShadowProps,
    space,
    SpaceProps,
    typography,
    TypographyProps,
} from 'styled-system';

import { Theme } from '../../../theme/theme.types';

export interface BoxProps
    extends ColorProps<Theme>,
        SpaceProps<Theme>,
        BorderProps<Theme>,
        PositionProps<Theme>,
        ShadowProps<Theme>,
        LayoutProps<Theme>,
        FlexboxProps<Theme>,
        TypographyProps<Theme>,
        Omit<React.HTMLAttributes<HTMLElement>, 'color'> {
    /**
     * Semantic element name. By default, the Text component renders text in a `span` tag.
     * @default span
     */
    as?: React.ElementType;
    children?: React.ReactNode;
}

const StyledBox = styled.div<BoxProps>`
    ${compose(flexbox, layout, shadow, space, border, color, position, typography)}
`;

export function Box({ children, ...props }: BoxProps) {
    return <StyledBox {...props}>{children}</StyledBox>;
}
```

**Comment by merveillevaneck** [comment]
> Adding the as prop fixed it for me. But don't ask me why clown_face

this works as in the above snippet. color prop seems to no longer be conflicting on the number instance.
